In [1]:
!pip install pandas
!pip install sklearn

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Preparing metadata (setup.py) ... done
  Created wheel for sklearn: filename=sklearn-0.0.post4-py3-none-any.whl size=2973 sha256=b7cc9f5b58356c4df0f9afcf80c7cc479bff4c2ff4436794470150eeadd08d61
  Stored in directory: /root/.cache/pip/wheels/46/b8/69/4f7789b7d2fc0718a44cbc4ae5d40970a40c22c472ae50f0a1
Successfully built sklearn


In [2]:
!pip install kaggle
!mkdir ~/.kaggle
!cp kaggle.json ~/.kaggle/
!chmod 600 ~/.kaggle/kaggle.json
!kaggle datasets download vora1011/ipl-2008-to-2021-all-match-dataset
!unzip ipl-2008-to-2021-all-match-dataset.zip

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  0% 0.00/1.58M [00:00<?, ?B/s]
100% 1.58M/1.58M [00:00<00:00, 117MB/s]
Archive:  ipl-2008-to-2021-all-match-dataset.zip
  inflating: IPL_Ball_by_Ball_2008_2022.csv  
  inflating: IPL_Matches_2008_2022.csv  


In [3]:
import pandas as pd

df=pd.read_csv('IPL_Ball_by_Ball_2008_2022.csv')
df.isnull().sum()

ID                        0
innings                   0
overs                     0
ballnumber                0
batter                    0
bowler                    0
non-striker               0
extra_type           213905
batsman_run               0
extras_run                0
total_run                 0
non_boundary              0
isWicketDelivery          0
player_out           214803
kind                 214803
fielders_involved    217966
BattingTeam               0
dtype: int64

In [6]:
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import accuracy_score, recall_score, precision_score, f1_score

# Select the features and target variable
features = ['innings', 'overs', 'ballnumber']
target = 'total_run'

# Prepare the data for classification
X = df[features].astype(str).apply(' '.join, axis=1)
y = df[target]

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Create a CountVectorizer to convert text data into numerical features
vectorizer = CountVectorizer()

# Fit and transform the training data
X_train_counts = vectorizer.fit_transform(X_train)

# Transform the testing data
X_test_counts = vectorizer.transform(X_test)

# Create a Naive Bayes classifier
naive_bayes_classifier = MultinomialNB()

# Train the classifier
naive_bayes_classifier.fit(X_train_counts, y_train)

# Predict on the test data
y_pred = naive_bayes_classifier.predict(X_test_counts)

# Calculate the accuracy
accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred, average='weighted')
recall = recall_score(y_test, y_pred, average='weighted')
f1 = f1_score(y_test, y_pred, average='weighted')

print("Accuracy:", accuracy)
print("Precision:", precision)
print("Recall:", recall)
print("F1-Score:", f1)


Accuracy: 0.4153481887986546
Precision: 0.1725141179383228
Recall: 0.4153481887986546
F1-Score: 0.24377622312817956


/usr/local/lib/python3.10/dist-packages/sklearn/metrics/_classification.py:1344: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Accuracy: The accuracy of the classifier is approximately 0.4153, which means that it correctly predicted the total runs around 41.53% of the time. However, it's important to note that accuracy alone might not provide a complete picture of the classifier's performance, especially if the dataset is imbalanced.

Precision: The precision score is approximately 0.1725. Precision represents the proportion of correctly predicted positive instances (total runs) out of all instances predicted as positive by the classifier. A lower precision indicates a higher number of false positives.

Recall: The recall score is also approximately 0.4153. Recall represents the proportion of correctly predicted positive instances (total runs) out of all actual positive instances in the dataset. A higher recall indicates a lower number of false negatives.

F1-Score: The F1-score is approximately 0.2438. The F1-score is the harmonic mean of precision and recall. It provides a balanced measure that considers both precision and recall. A higher F1-score indicates a better balance between precision and recall.

Based on these metrics, it seems that the Naive Bayes classifier might not be performing well on this particular classification task, as the scores are relatively low.